File: Domain/AppMetaData/Router.cs
namespace Domain.AppMetaData
{
    public static class Router
    {
        public const string SingleRoute = "{id}";

        public const string Root = "api";
        public const string Version = "v1";
        public const string Rule = Root + "/" + Version + "/";

        public static class Authentication
        {
            public const string Prefix = Rule + "authenticate/";
            public const string SignIn = Prefix + "signIn";
            public const string SignInViaGoogle = Prefix + "signInViaGoogle";
            public const string RefreshToken = Prefix + "refreshToken";
            public const string ValidateToken = Prefix + "validateToken";
            public const string SendResetPasswordCode = Prefix + "sendResetPasswordCode";
            public const string ConfirmResetPasswordCode = Prefix + "confirmResetPasswordCode";
            public const string ResetPassword = Prefix + "resetPassword";
            public const string ConfirmEmail = "api/authenticate/confirmEmail";
            public const string TwoStepVerification = Prefix + "twoStepVerification";
            public const string Logout = Prefix + "logout";
            public const string LogoutAll = Prefix + "logoutAll";
        }

        public static class Authorization
        {
            public const string Prefix = Rule + "authorize/";
            public const string Role = Prefix + "role/";
            public const string Claim = Prefix + "claim/";
            public const string CreateRole = Role + "create";
            public const string EditRole = Role + "edit";
            public const string DeleteRole = Role + SingleRoute;
            public const string GetAllRoles = Role + "getAll";
            public const string GetRoleById = Role + SingleRoute;
            public const string ManageUserRoles = Role + "manageUserRoles/" + SingleRoute;
            public const string UpdateUserRoles = Role + "updateUserRoles";
            public const string ManageUserClaims = Claim + "manageUserClaims/" + SingleRoute;
            public const string UpdateUserClaims = Claim + "updateUserClaims";
        }

        public static class UserRouting
        {
            public const string Prefix = Rule + "user/";
            public const string Register = Prefix + "register";
            public const string Profile = Prefix + "profile";
            public const string ChangePassword = Prefix + "changePassword";
            public const string CreateAdmin = Prefix + "createAdmin";
            public const string CreateVendor = Prefix + "createVendor";

            public const string MyActivitiesPaginated = Prefix + "activities/paginated";

            public const string Documents = Prefix + "documents";
            public const string DocumentById = Prefix + "documents/" + SingleRoute;
            public const string DocumentDownload = Prefix + "documents/" + SingleRoute + "/download";

            // Unified Users Management
            public const string UsersPaginated = Prefix + "users/paginated";
            public const string GetUserById = Prefix + "users/" + SingleRoute;
            public const string CreateUser = Prefix + "users/create";
            public const string EditUser = Prefix + "users/edit";
            public const string DeleteUser = Prefix + "users/" + SingleRoute;
            public const string ToggleUserStatus = Prefix + "users/" + SingleRoute + "/toggleStatus";
        }

        public static class CategoryRouting
        {
            public const string Prefix = Rule + "category/";
            public const string GetAll = Prefix + "getAll";
            public const string Paginated = Prefix + "paginated";
            public const string GetById = Prefix + "getById";
            public const string Create = Prefix + "create";
            public const string Edit = Prefix + "edit";
            public const string Delete = Prefix + "delete";
        }

        public static class SubCategoryRouting
        {
            public const string Prefix = Rule + "subCategory/";
            public const string GetAll = Prefix + "getAll";
            public const string Paginated = Prefix + "paginated";
            public const string GetById = Prefix + SingleRoute;
            public const string Create = Prefix + "create";
            public const string Edit = Prefix + "edit";
            public const string Delete = Prefix + SingleRoute;
        }

        public static class BrandRouting
        {
            public const string Prefix = Rule + "brand/";
            public const string GetAll = Prefix + "getAll";
            public const string Paginated = Prefix + "paginated";
            public const string GetById = Prefix + SingleRoute;
            public const string Create = Prefix + "create";
            public const string Edit = Prefix + "edit";
            public const string Delete = Prefix + SingleRoute;
        }

        public static class TagRouting
        {
            public const string Prefix = Rule + "tag/";
            public const string GetAll = Prefix + "getAll";
            public const string Paginated = Prefix + "paginated";
            public const string GetById = Prefix + "getById";
            public const string Create = Prefix + "create";
            public const string Edit = Prefix + "edit";
            public const string Delete = Prefix + "delete";
        }

        public static class UnitOfMeasureRouting
        {
            public const string Prefix = Rule + "unit/";
            public const string GetAll = Prefix + "getAll";
            public const string Paginated = Prefix + "paginated";
            public const string GetById = Prefix + SingleRoute;
            public const string Create = Prefix + "create";
            public const string Edit = Prefix + "edit";
            public const string Delete = Prefix + SingleRoute;
        }

        public static class WarrantyRouting
        {
            public const string Prefix = Rule + "warranty/";
            public const string GetAll = Prefix + "getAll";
            public const string Paginated = Prefix + "paginated";
            public const string GetById = Prefix + "getById";
            public const string Create = Prefix + "create";
            public const string Edit = Prefix + "edit";
            public const string Delete = Prefix + "delete";
        }

        public static class VariantAttributeRouting
        {
            public const string Prefix = Rule + "variantAttribute/";
            public const string GetAll = Prefix + "getAll";
            public const string Paginated = Prefix + "paginated";
            public const string GetById = Prefix + SingleRoute;
            public const string Create = Prefix + "create";
            public const string Edit = Prefix + "edit";
            public const string Delete = Prefix + SingleRoute;
        }

        public static class CouponRouting
        {
            public const string Prefix = Rule + "coupon/";
            public const string GetAll = Prefix + "getAll";
            public const string Paginated = Prefix + "paginated";
            public const string GetById = Prefix + SingleRoute;
            public const string Create = Prefix + "create";
            public const string Edit = Prefix + "edit";
            public const string Delete = Prefix + SingleRoute;
        }

        public static class GiftCardRouting
        {
            public const string Prefix = Rule + "giftCard/";
            public const string GetAll = Prefix + "getAll";
            public const string Paginated = Prefix + "paginated";
            public const string GetById = Prefix + SingleRoute;
            public const string Create = Prefix + "create";
            public const string Edit = Prefix + "edit";
            public const string Delete = Prefix + SingleRoute;
        }

        public static class DiscountRouting
        {
            public const string Prefix = Rule + "discount/";
            public const string GetAll = Prefix + "getAll";
            public const string Paginated = Prefix + "paginated";
            public const string GetById = Prefix + SingleRoute;
            public const string Create = Prefix + "create";
            public const string Edit = Prefix + "edit";
            public const string Delete = Prefix + SingleRoute;
        }

        public static class AccountRouting
        {
            public const string Prefix = Rule + "account/";
            public const string GetAll = Prefix + "getAll";
            public const string Paginated = Prefix + "paginated";
            public const string GetById = Prefix + SingleRoute;
            public const string Create = Prefix + "create";
            public const string Edit = Prefix + "edit";
            public const string Delete = Prefix + SingleRoute;
        }

        public static class ProductRouting
        {
            public const string Prefix = Rule + "product/";
            public const string Paginated = Prefix + "paginated";
            public const string GetSingle = Prefix + "getSingle";
            public const string Create = Prefix + "create";
            public const string Edit = Prefix + "edit";
            public const string Delete = Prefix + SingleRoute;
        }

        public static class OrderRouting
        {
            public const string Prefix = Rule + "order/";
            public const string GetMyOrders = Prefix + "getMyOrders";
            public const string Paginated = Prefix + "paginated";
            public const string GetById = Prefix + SingleRoute;
            public const string Create = Prefix + "create";
            public const string Delete = Prefix + SingleRoute;
            public const string PlaceOrder = Prefix + "placeOrder/" + SingleRoute;
        }

        public static class CartRouting
        {
            public const string Prefix = Rule + "cart/";
            public const string GetById = Prefix + SingleRoute;
            public const string GetMyCart = Prefix + "myCart";
            public const string AddToCart = Prefix + "addToCart";
            public const string UpdateItemQuantity = Prefix + "updateItemQuantity";
            public const string RemoveFromCart = Prefix + "removeFromCart/" + SingleRoute;
            public const string Delete = Prefix + SingleRoute;
        }

        public static class CustomerRouting
        {
            public const string Prefix = Rule + "customer/";
            public const string Paginated = Prefix + "paginated";
            public const string GetById = Prefix + SingleRoute;
            public const string Create = Prefix + "create";
            public const string Edit = Prefix + "edit";
            public const string Delete = Prefix + SingleRoute;
            public const string ToggleStatus = Prefix + "toggleStatus/" + SingleRoute;
        }

        public static class VendorRouting
        {
            public const string Prefix = Rule + "vendor/";
            public const string Paginated = Prefix + "paginated";
            public const string GetById = Prefix + SingleRoute;
            public const string Create = Prefix + "create";
            public const string Edit = Prefix + "edit";
            public const string Delete = Prefix + SingleRoute;
            public const string ToggleStatus = Prefix + "toggleStatus";
        }

        public static class AdminRouting
        {
            public const string Prefix = Rule + "admin/";
            public const string Paginated = Prefix + "paginated";
            public const string GetById = Prefix + SingleRoute;
            public const string Create = Prefix + "create";
            public const string Edit = Prefix + "edit";
            public const string Delete = Prefix + SingleRoute;
            public const string ToggleStatus = Prefix + "toggleStatus";
        }

        public static class EmployeeRouting
        {
            public const string Prefix = Rule + "employee/";
            public const string GetAll = Prefix + "getAll";
            public const string Paginated = Prefix + "paginated";
            public const string GetById = Prefix + SingleRoute;
            public const string Create = Prefix + "create";
            public const string Edit = Prefix + "edit";
            public const string Delete = Prefix + SingleRoute;
        }

        public static class PaymentRouting
        {
            public const string Prefix = Rule + "payment/";
            public const string SetPaymentMethod = Prefix + "setPaymentMethod";
            public const string ServerCallback = Prefix + "serverCallback";
            public const string PaymobCallback = Prefix + "paymobCallback";
        }

        public static class DeliveryRouting
        {
            public const string Prefix = Rule + "delivery/";
            public const string SetDeliveryMethod = Prefix + "setDeliveryMethod";
            public const string EditDeliveryMethod = Prefix + "editDeliveryMethod";
        }

        public static class ShippingAddressRouting
        {
            public const string Prefix = Rule + "shippingAddress/";
            public const string GetAll = Prefix + "getAll";
            public const string GetById = Prefix + SingleRoute;
            public const string Create = Prefix + "create";
            public const string Edit = Prefix + "edit";
            public const string Delete = Prefix + SingleRoute;
            public const string SetShippingAddress = Prefix + "setShippingAddress";
        }

        public static class ReviewRouting
        {
            public const string Prefix = Rule + "review/";
            public const string Paginated = Prefix + "paginated";
            public const string Create = Prefix + "create";
            public const string Edit = Prefix + "edit";
            public const string Delete = Prefix + SingleRoute;
        }

        public static class NotificationsRouting
        {
            public const string Prefix = Rule + "notifications/";
            public const string List = Prefix;
            public const string Paginated = Prefix + "paginated";
            public const string MarkAsRead = Prefix + "markAsRead/" + SingleRoute;
            public const string MarkAllAsRead = Prefix + "markAllAsRead";
            public const string MarkRead = Prefix + "mark-read";
        }

        public static class AuditLogRouting
        {
            public const string Prefix = Rule + "auditLog/";
            public const string Paginated = Prefix + "paginated";
            public const string GetById = Prefix + SingleRoute;
        }

        public static class LookUpsRouting
        {
            public const string Prefix = Rule + "lookups/";
            public const string Categories = Prefix + "categories";
            public const string SubCategories = Prefix + "subCategories";
            public const string SubCategoriesByCategory = Prefix + "subCategoriesByCategory";
            public const string Brands = Prefix + "brands";
            public const string UnitOfMeasures = Prefix + "unitOfMeasures";
            public const string Warranties = Prefix + "warranties";
            public const string VariantAttributes = Prefix + "variantAttributes";
            public const string Roles = Prefix + "roles";

            public const string ProductPublishStatuses = Prefix + "productPublishStatuses";
            public const string ProductVisibilities = Prefix + "productVisibilities";
            public const string ProductTypes = Prefix + "productTypes";
            public const string SellingTypes = Prefix + "sellingTypes";
            public const string TaxTypes = Prefix + "taxTypes";
            public const string DiscountTypes = Prefix + "discountTypes";
            public const string Tags = Prefix + "tags";
        }
    }
}

File: client/src/config/Router.ts
import { API_URL } from '@/config/apiConfig';

export const SingleRoute = "{id}";
export const Root = "api";
export const Version = "v1";
export const Rule = `${API_URL}/${Root}/${Version}/`;

export const Router = {
    SingleRoute,
    Root,
    Version,
    Rule,

    Authentication: {
        Prefix: `${Rule}authenticate/`,
        SignIn: `${Rule}authenticate/signIn`,
        SignInViaGoogle: `${Rule}authenticate/signInViaGoogle`,
        RefreshToken: `${Rule}authenticate/refreshToken`,
        ValidateToken: `${Rule}authenticate/validateToken`,
        SendResetPasswordCode: `${Rule}authenticate/sendResetPasswordCode`,
        ConfirmResetPasswordCode: `${Rule}authenticate/confirmResetPasswordCode`,
        ResetPassword: `${Rule}authenticate/resetPassword`,
        ConfirmEmail: `${API_URL}/api/authenticate/confirmEmail`,
        TwoStepVerification: `${Rule}authenticate/twoStepVerification`,
        Logout: `${Rule}authenticate/logout`,
        LogoutAll: `${Rule}authenticate/logoutAll`,
        Register: `${Rule}authenticate/register`,
        ChangePassword: `${Rule}authenticate/changePassword`,
    },

    Authorization: {
        Prefix: `${Rule}authorize/`,
        Role: `${Rule}authorize/role/`,
        Claim: `${Rule}authorize/claim/`,
        CreateRole: `${Rule}authorize/role/create`,
        EditRole: `${Rule}authorize/role/edit`,
        DeleteRole: `${Rule}authorize/role/${SingleRoute}`,
        GetAllRoles: `${Rule}authorize/role/getAll`,
        GetRoleById: `${Rule}authorize/role/${SingleRoute}`,
        ManageUserRoles: `${Rule}authorize/role/manageUserRoles/${SingleRoute}`,
        UpdateUserRoles: `${Rule}authorize/role/updateUserRoles`,
        ManageUserClaims: `${Rule}authorize/claim/manageUserClaims/${SingleRoute}`,
        UpdateUserClaims: `${Rule}authorize/claim/updateUserClaims`,
    },

    UserRouting: {
        Prefix: `${Rule}user/`,
        Profile: `${Rule}user/profile`,
        CreateAdmin: `${Rule}user/createAdmin`,
        CreateVendor: `${Rule}user/createVendor`,
        UsersPaginated: `${Rule}user/users/paginated`,
        GetUserById: `${Rule}user/users/${SingleRoute}`,
        CreateUser: `${Rule}user/users/create`,
        EditUser: `${Rule}user/users/edit`,
        DeleteUser: `${Rule}user/users/${SingleRoute}`,
        ToggleUserStatus: `${Rule}user/users/${SingleRoute}/toggleStatus`,
    },

    CategoryRouting: {
        Prefix: `${Rule}category/`,
        GetAll: `${Rule}category/getAll`,
        Paginated: `${Rule}category/paginated`,
        GetById: `${Rule}category/getById`,
        Create: `${Rule}category/create`,
        Edit: `${Rule}category/edit`,
        Delete: `${Rule}category/delete`,
    },

    SubCategoryRouting: {
        Prefix: `${Rule}subCategory/`,
        GetAll: `${Rule}subCategory/getAll`,
        Paginated: `${Rule}subCategory/paginated`,
        GetById: `${Rule}subCategory/${SingleRoute}`,
        Create: `${Rule}subCategory/create`,
        Edit: `${Rule}subCategory/edit`,
        Delete: `${Rule}subCategory/${SingleRoute}`,
    },

    BrandRouting: {
        Prefix: `${Rule}brand/`,
        GetAll: `${Rule}brand/getAll`,
        Paginated: `${Rule}brand/paginated`,
        GetById: `${Rule}brand/${SingleRoute}`,
        Create: `${Rule}brand/create`,
        Edit: `${Rule}brand/edit`,
        Delete: `${Rule}brand/${SingleRoute}`,
    },

    TagRouting: {
        Prefix: `${Rule}tag/`,
        GetAll: `${Rule}tag/getAll`,
        Paginated: `${Rule}tag/paginated`,
        GetById: `${Rule}tag/getById`,
        Create: `${Rule}tag/create`,
        Edit: `${Rule}tag/edit`,
        Delete: `${Rule}tag/delete`,
    },

    UnitOfMeasureRouting: {
        Prefix: `${Rule}unit/`,
        GetAll: `${Rule}unit/getAll`,
        Paginated: `${Rule}unit/paginated`,
        GetById: `${Rule}unit/${SingleRoute}`,
        Create: `${Rule}unit/create`,
        Edit: `${Rule}unit/edit`,
        Delete: `${Rule}unit/${SingleRoute}`,
    },

    WarrantyRouting: {
        Prefix: `${Rule}warranty/`,
        GetAll: `${Rule}warranty/getAll`,
        Paginated: `${Rule}warranty/paginated`,
        GetById: `${Rule}warranty/getById`,
        Create: `${Rule}warranty/create`,
        Edit: `${Rule}warranty/edit`,
        Delete: `${Rule}warranty/delete`,
    },

    VariantAttributeRouting: {
        Prefix: `${Rule}variantAttribute/`,
        GetAll: `${Rule}variantAttribute/getAll`,
        Paginated: `${Rule}variantAttribute/paginated`,
        GetById: `${Rule}variantAttribute/${SingleRoute}`,
        Create: `${Rule}variantAttribute/create`,
        Edit: `${Rule}variantAttribute/edit`,
        Delete: `${Rule}variantAttribute/${SingleRoute}`,
    },

    CouponRouting: {
        Prefix: `${Rule}coupon/`,
        GetAll: `${Rule}coupon/getAll`,
        Paginated: `${Rule}coupon/paginated`,
        GetById: `${Rule}coupon/${SingleRoute}`,
        Create: `${Rule}coupon/create`,
        Edit: `${Rule}coupon/edit`,
        Delete: `${Rule}coupon/${SingleRoute}`,
    },

    GiftCardRouting: {
        Prefix: `${Rule}giftCard/`,
        GetAll: `${Rule}giftCard/getAll`,
        Paginated: `${Rule}giftCard/paginated`,
        GetById: `${Rule}giftCard/${SingleRoute}`,
        Create: `${Rule}giftCard/create`,
        Edit: `${Rule}giftCard/edit`,
        Delete: `${Rule}giftCard/${SingleRoute}`,
    },

    DiscountRouting: {
        Prefix: `${Rule}discount/`,
        GetAll: `${Rule}discount/getAll`,
        Paginated: `${Rule}discount/paginated`,
        GetById: `${Rule}discount/${SingleRoute}`,
        Create: `${Rule}discount/create`,
        Edit: `${Rule}discount/edit`,
        Delete: `${Rule}discount/${SingleRoute}`,
    },

    AccountRouting: {
        Prefix: `${Rule}account/`,
        GetAll: `${Rule}account/getAll`,
        Paginated: `${Rule}account/paginated`,
        GetById: `${Rule}account/${SingleRoute}`,
        Create: `${Rule}account/create`,
        Edit: `${Rule}account/edit`,
        Delete: `${Rule}account/${SingleRoute}`,
    },

    ProductRouting: {
        Prefix: `${Rule}product/`,
        Paginated: `${Rule}product/paginated`,
        GetSingle: `${Rule}product/getSingle`,
        Create: `${Rule}product/create`,
        Edit: `${Rule}product/edit`,
        Delete: `${Rule}product/${SingleRoute}`,
    },

    OrderRouting: {
        Prefix: `${Rule}order/`,
        GetMyOrders: `${Rule}order/getMyOrders`,
        Paginated: `${Rule}order/paginated`,
        GetById: `${Rule}order/${SingleRoute}`,
        Create: `${Rule}order/create`,
        Delete: `${Rule}order/${SingleRoute}`,
        PlaceOrder: `${Rule}order/placeOrder/${SingleRoute}`,
    },

    CartRouting: {
        Prefix: `${Rule}cart/`,
        GetById: `${Rule}cart/${SingleRoute}`,
        GetMyCart: `${Rule}cart/myCart`,
        AddToCart: `${Rule}cart/addToCart`,
        UpdateItemQuantity: `${Rule}cart/updateItemQuantity`,
        RemoveFromCart: `${Rule}cart/removeFromCart/${SingleRoute}`,
        Delete: `${Rule}cart/${SingleRoute}`,
    },

    CustomerRouting: {
        Prefix: `${Rule}customer/`,
        Paginated: `${Rule}customer/paginated`,
        GetById: `${Rule}customer/${SingleRoute}`,
        Create: `${Rule}customer/create`,
        Edit: `${Rule}customer/edit`,
        Delete: `${Rule}customer/${SingleRoute}`,
        ToggleStatus: `${Rule}customer/toggleStatus/${SingleRoute}`,
    },

    VendorRouting: {
        Prefix: `${Rule}vendor/`,
        Paginated: `${Rule}vendor/paginated`,
        GetById: `${Rule}vendor/${SingleRoute}`,
        Create: `${Rule}vendor/create`,
        Edit: `${Rule}vendor/edit`,
        Delete: `${Rule}vendor/${SingleRoute}`,
        ToggleStatus: `${Rule}vendor/toggleStatus`,
    },

    AdminRouting: {
        Prefix: `${Rule}admin/`,
        Paginated: `${Rule}admin/paginated`,
        GetById: `${Rule}admin/${SingleRoute}`,
        Create: `${Rule}admin/create`,
        Edit: `${Rule}admin/edit`,
        Delete: `${Rule}admin/${SingleRoute}`,
        ToggleStatus: `${Rule}admin/toggleStatus`,
    },

    EmployeeRouting: {
        Prefix: `${Rule}employee/`,
        GetAll: `${Rule}employee/getAll`,
        Paginated: `${Rule}employee/paginated`,
        GetById: `${Rule}employee/${SingleRoute}`,
        Create: `${Rule}employee/create`,
        Edit: `${Rule}employee/edit`,
        Delete: `${Rule}employee/${SingleRoute}`,
    },

    PaymentRouting: {
        Prefix: `${Rule}payment/`,
        SetPaymentMethod: `${Rule}payment/setPaymentMethod`,
        ServerCallback: `${Rule}payment/serverCallback`,
        PaymobCallback: `${Rule}payment/paymobCallback`,
    },

    DeliveryRouting: {
        Prefix: `${Rule}delivery/`,
        SetDeliveryMethod: `${Rule}delivery/setDeliveryMethod`,
        EditDeliveryMethod: `${Rule}delivery/editDeliveryMethod`,
    },

    ShippingAddressRouting: {
        Prefix: `${Rule}shippingAddress/`,
        GetAll: `${Rule}shippingAddress/getAll`,
        GetById: `${Rule}shippingAddress/${SingleRoute}`,
        Create: `${Rule}shippingAddress/create`,
        Edit: `${Rule}shippingAddress/edit`,
        Delete: `${Rule}shippingAddress/${SingleRoute}`,
        SetShippingAddress: `${Rule}shippingAddress/setShippingAddress`,
    },

    ReviewRouting: {
        Prefix: `${Rule}review/`,
        Paginated: `${Rule}review/paginated`,
        Create: `${Rule}review/create`,
        Edit: `${Rule}review/edit`,
        Delete: `${Rule}review/${SingleRoute}`,
    },

    NotificationsRouting: {
        Prefix: `${Rule}notifications/`,
        List: `${Rule}notifications/`,
        Paginated: `${Rule}notifications/paginated`,
        MarkAsRead: `${Rule}notifications/markAsRead/${SingleRoute}`,
        MarkAllAsRead: `${Rule}notifications/markAllAsRead`,
        MarkRead: `${Rule}notifications/mark-read`,
    },

    AuditLogRouting: {
        Prefix: `${Rule}auditLog/`,
        Paginated: `${Rule}auditLog/paginated`,
        GetById: `${Rule}auditLog/${SingleRoute}`,
    },

    LookUpsRouting: {
        Prefix: `${Rule}lookups/`,
        Categories: `${Rule}lookups/categories`,
        SubCategories: `${Rule}lookups/subCategories`,
        SubCategoriesByCategory: `${Rule}lookups/subCategoriesByCategory`,
        Brands: `${Rule}lookups/brands`,
        UnitOfMeasures: `${Rule}lookups/unitOfMeasures`,
        Warranties: `${Rule}lookups/warranties`,
        VariantAttributes: `${Rule}lookups/variantAttributes`,
        Roles: `${Rule}lookups/roles`,
        ProductPublishStatuses: `${Rule}lookups/productPublishStatuses`,
        ProductVisibilities: `${Rule}lookups/productVisibilities`,
        ProductTypes: `${Rule}lookups/productTypes`,
        SellingTypes: `${Rule}lookups/sellingTypes`,
        TaxTypes: `${Rule}lookups/taxTypes`,
        DiscountTypes: `${Rule}lookups/discountTypes`,
        Tags: `${Rule}lookups/tags`,
    },
};
